src/main/java/io/debezium/connector/yugabytedb/YBPartition.java

```diff
@@ -11,6 +11,10 @@ public class YBPartition implements Partition {
 
     private final String tabletId;
 
+    public YBPartition() {
+        this.tabletId = "";
```

Comment:
With this, there is a possibility that in a certain code flow if we have a condition where we check for tabletId to be null then having this will bypass that condition. We need to be sure that we are not violating that. eg.
```java
if (partition.getTabletId() == null ) {
  // do something
} else {
  // do something else - with the above change, we will end up coming to this block all the time
}
```

Reply:
This is a good point. I checked PostgresConnector and checked that it shouldn't have the default constructor. I removed and then have fixed a bunch of APIs to calculate previous offsets. Thanks for pointing this out.